Sources/XCHammer/XcodeBuildSystemInstaller.swift

```diff
@@ -86,6 +55,29 @@ enum XcodeBuildSystemInstaller {
         return .success(())
     }
 
+    static func uninstallIfNecessary() -> Result<(), CommandError> {
+        do {
+            let needsUninstalls = try getXcodes(withBuildSystemInstalled: true)
+
+            for xcode in needsUninstalls {
```

Comment:
A few edge cases for some of the logic here:
1. what happens if the person installs a new Xcode then trys to run uninstall? 
2. It looks like it isn't fully atomic - e.g. for some reason one install is corrupted or program is deleted
    2a. it looks on [line 70](https://github.com/bazel-ios/xchammer/pull/32/files#diff-690a96f8d32336af23e9548d935904b66ba45ab2ed1e9564fba5b0be9a9b3f58R70) it can throw an

Reply:
`getXcodes` contains logic to list existing Xcodes and filter only Xcode with the build system installed (hence `withBuildSystemInstalled: true`). This won't return any Xcode that don't satisfy the conditions to be installed in it's impl, i.e., a symlink exists and points to the path from the pkg `/opt/XCBuildKit/XCBuildKit.app/Contents/MacOS/BazelBuildService`.

Tested this before submitting, If one Xcode has it installed and the other one is untouched this installation won't break and will uninstall only from the correct one since `getXcodes` will return only the Xcode that is eligible for uninstallation.

That said the remove step shouldn't thrown unless there's bad logic returning wrong results in `getXcodes`. I can add a check to be extra safe, can also assert that the `.default` original service exists on disk before calling a `mv`.

Will push that soon.

Reply:
https://github.com/bazel-ios/xchammer/pull/32/commits/ae43b2402ae8976aea089d0273ef5e994ffa0e24